pkg/minishift/addon/addon_meta.go

```diff
@@ -86,11 +94,36 @@ func (meta *DefaultAddOnMeta) Description() []string {
 	return meta.headers[requiredMetaTags[1]].([]string)
 }
 
-func (meta *DefaultAddOnMeta) RequiredVars() []string {
+func (meta *DefaultAddOnMeta) RequiredVars() ([]string, error) {
 	if val, contains := meta.headers[requiredVars].(string); contains {
 		return meta.splitAndTrim(val)
 	} else {
-		return []string{}
+		return []string{}, nil
+	}
+}
+
+func (meta *DefaultAddOnMeta) VarDefaults() ([]RequiredVar, error) {
+	if val, contains := meta.headers[varDefaults].(string); contains {
```

Comment:
-1 Basically ok, but you want to do this as part of _NewAddOnMeta_. There is no need to defer this until _VarDefaults()_ is called. Also, each time you call VarDefault(), you will have to re-create it. If you do it as part of __NewAddOnMeta__ you also get this fail early approach I was talking about. You will get an error already when the add-on is read and the metadata is created. Now I would think you won't get an error until you apply the add-on.

Reply:
Done and looking nice now. Thanks for suggestion :smile: